packages/promptable/src/prompts/Prompt.ts

```diff
@@ -1,17 +1,18 @@
 import { injectVariables } from "@utils/inject-variables";
+import { ExtractVariableNames } from "@utils/extract-variable-names";
 import { NoopParser, Parser } from "@prompts/Parser";
 
 export class Prompt<
   T extends string = string,
   P extends Parser<any> = NoopParser
 > implements Parser<any>
 {
-  text: string;
-  variableNames: T[];
+  text: T;
+  variableNames: ExtractVariableNames<T>;
```

Comment:
This might be wrong. I think I want `Array<keyof  ExtractVariableNames<T>>` here. Still working on this pr.